From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 12:45:32 -0700
Subject: [PATCH] mm/damon: remove unused damon_access_report fields

Those may be used in future, but not for now.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 6 ------
 mm/memory.c           | 1 -
 2 files changed, 7 deletions(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index cc60508ce7c4..f7139ba81103 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -111,9 +111,6 @@ struct damon_target {
  *			NULL if it is of the physical address.
  * @addr:		The start address of the reporting region.
  * @size:		The size of the reporting region.
- * @nr_accesses:	Number of detected accesses to the region.
- * @node_id:		NUMA node that made the access.
- * @write:		Whether the access is write.
  *
  * @pid could be stale, and hence shouldn't be de-referenced.
  */
@@ -121,9 +118,6 @@ struct damon_access_report {
 	struct pid *pid;
 	unsigned long addr;
 	unsigned long size;
-	int nr_accesses;
-	int node_id;
-	bool write;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
diff --git a/mm/memory.c b/mm/memory.c
index 3f133401847a..adeb0d7f6a6b 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -5978,7 +5978,6 @@ static void do_damon_page(struct vm_fault *vmf, bool huge_pmd)
 	struct damon_access_report access_report = {
 		.addr = vmf->address,
 		.size = 1,
-		.nr_accesses = 1,
 	};
 
 	if (huge_pmd)
-- 
2.39.5

